### PR TITLE
feat(routeFromHAR): add arguments

### PR DESCRIPTION
**Pull request by NoamGaash**
When using `routeFromHAR` to record traffic for specific test cases, the user has less control over the recording mechanism (compared to recording HARs in the browser context).
This commit suggests passing more arguments, giving the user more control.

### CODE REVIEW

**Comment by NoamGaash** [comment]
@microsoft-github-policy-service agree

**Comment by pavelfeldman** [comment]
Your change looks good, but I'd like to better understand your use case. Could it be that you don't really need routeFromHAR and you use it to get a manual control on when har starts / stops recording? We might expose a different API for that on sat context.tracing.startHAR / stopHAR.

File: docs/src/api/class-browsercontext.md [hunk 1]
```diff
@@ -1124,6 +1124,18 @@ If specified, updates the given HAR with the actual network information instead
 
 A glob pattern, regular expression or predicate to match the request URL. Only requests with URL matching the pattern will be served from the HAR file. If not specified, all requests are served from the HAR file.
 
+### option: BrowserContext.routeFromHAR.mode
+* since: v1.32
+- `mode` <[HarMode]<"full"|"minimal">>
+
+When set to `minimal`, only record information necessary for routing from HAR. This omits sizes, timing, page, cookies, security and other types of HAR information that are not used when replaying from HAR. Defaults to `full`.
+
+### option: BrowserContext.routeFromHAR.url
```
Comment (pavelfeldman) [inline docs/src/api/class-browsercontext.md:1133]:
```suggestion
### option: BrowserContext.routeFromHAR.content
```

File: docs/src/api/class-browsercontext.md [hunk 2]
```diff
@@ -1124,6 +1124,18 @@ If specified, updates the given HAR with the actual network information instead
 
 A glob pattern, regular expression or predicate to match the request URL. Only requests with URL matching the pattern will be served from the HAR file. If not specified, all requests are served from the HAR file.
 
+### option: BrowserContext.routeFromHAR.mode
+* since: v1.32
+- `mode` <[HarMode]<"full"|"minimal">>
+
+When set to `minimal`, only record information necessary for routing from HAR. This omits sizes, timing, page, cookies, security and other types of HAR information that are not used when replaying from HAR. Defaults to `full`.
```
Comment (pavelfeldman) [inline docs/src/api/class-browsercontext.md:1131]:
```suggestion
When set to `minimal`, only record information necessary for routing from HAR. This omits sizes, timing, page, cookies, security and other types of HAR information that are not used when replaying from HAR. Defaults to `minimal`.
```

**Comment by NoamGaash** [comment]
Thanks!
In my company, we love using Playwright to test our client side.
We want to isolate some of our tests from the backend and mock the network traffic using HAR files.

We have some tests that change the state on the server side, and in order to mock this behavior and make sequential GET requests serve different outputs, we have used [this solution](https://github.com/microsoft/playwright/issues/18288#issuecomment-1324056041) by @olivierbeaulieu.

our code seems like this:
```js
await harCheckpoint('test-checkpoint-1')
// go to the page
//get data
await harCheckpoint('test-checkpoint2')
// perform action that changes the data
// get updated data
```
When the HAR recorder operates in `attach` mode, many different files are created, and it's very tricky to follow which file belongs to each checkpoint.

I want to record my HAR filed in `embed` mode and in this pull request I found a way to do it as a non-breaking change :)

Thank you!

**Comment by pavelfeldman** [comment]
Ok, sounds good. I was verifying you were using recordHAR as recordHAR, which it looks like what you are doing.

**Pull request merged**